daemon/config.go

```diff
@@ -121,6 +125,10 @@ type CommonConfig struct {
 	// may take place at a time for each push.
 	MaxConcurrentUploads *int `json:"max-concurrent-uploads,omitempty"`
 
+	// ShutdownTimeout is the timeout value (in seconds) the daemon will wait for the container
+	// to stop when daemon is being shutdown
+	ShutdownTimeout int `json:"shutdown-timeout,omitempty"`
```

Comment:
@cpuguy83 The `--shutdown-timeout` is the global timeout and is handled in `dockerd`. It could be unmarshaled from  `--config-file=/etc/docker/daemon.json`.

Another PR (#22566) is the per-container timeout which could be handled in client.

Maybe for this PR we use `seconds` , and for the other PR we could change the per-container timeout (in #22566) to `time.Duration`?